SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -204,24 +210,46 @@ bool ValidateIdentifierNames(TArray<TSharedPtr<FUnrealType>>& TypeInfos)
 		UClass* Class = Cast<UClass>(TypeInfo->Type);
 		check(Class);
 		const FString& ClassName = Class->GetName();
+		const FString& ClassPath = Class->GetPathName();
 		FString SchemaName = UnrealNameToSchemaName(ClassName);
 
-		if (!CheckSchemaNameValidity(SchemaName, Class->GetPathName(), TEXT("Class")))
+		if (!CheckSchemaNameValidity(SchemaName, ClassPath, TEXT("Class")))
 		{
 			bSuccess = false;
 		}
 
-		int Suffix = 0;
-		while (UsedSchemaNames.Contains(SchemaName))
+		FString DesiredSchemaName = SchemaName;
+
+		if (ClassPathToSchemaName.Contains(ClassPath))
 		{
-			UE_LOG(LogSpatialGDKSchemaGenerator, Warning, TEXT("Class name collision after removing non-alphanumeric characters. Name '%s' collides for types '%s' and '%s'. Will add a suffix to resolve the collision."),
-				*SchemaName, *Class->GetPathName(), *UsedSchemaNames[SchemaName]->GetPathName());
+			// Class already has generated schema name.
+			AddPotentialNameCollision(DesiredSchemaName, ClassPath, ClassPathToSchemaName[ClassPath]);
```

Comment:
Wouldn't we have already added it in `ResetUsedNames`?

Reply:
Yes, Done